fast64_internal/f3d/f3d_gbi.py

```diff
@@ -2451,26 +2453,41 @@ def __init__(self, name, index, fMaterial):
 		self.fMaterial = fMaterial
 		self.vertexList = VtxList(name + '_vtx_' + str(index))
 		self.triList = GfxList(name + '_tri_' + str(index), GfxListTag.Geometry, DLFormat.Static)
+		self.celTriLists = []
+		self.celTriListBaseName = name + '_tri_' + str(index) + '_cel'
+		
+	def add_cel_tri_list(self):
+		ret = GfxList(self.celTriListBaseName + str(len(self.celTriLists)), GfxListTag.Geometry, DLFormat.Static)
+		self.celTriLists.append(ret)
+		return ret
 	
 	def get_ptr_addresses(self, f3d):
 		return self.triList.get_ptr_addresses(f3d)
 	
 	def set_addr(self, startAddress, f3d):	
-		addrRange = self.triList.set_addr(startAddress, f3d)
+		addrRange = [0, startAddress]
+		for ctl in self.celTriLists:
+			addrRange = ctl.set_addr(addrRange[1], f3d)
+		addrRange = self.triList.set_addr(addrRange[1], f3d)
 		addrRange = self.vertexList.set_addr(addrRange[1])
 		return startAddress, addrRange[1]
 
 	def save_binary(self, romfile, f3d, segments):
+		for ctl in self.celTriLists:
+			ctl.save_binary(romfile, f3d, segments)
 		self.triList.save_binary(romfile, f3d, segments)
 		self.vertexList.save_binary(romfile)
 	
 	def to_c(self, f3d, gfxFormatter):
 		data = CData()
 		data.append(self.vertexList.to_c())
+		for ctl in self.celTriLists:
+			data.append(ctl.to_c(f3d))
 		data.append(self.triList.to_c(f3d))
 		return data
 
 	def to_c_vertex_scroll(self, gfxFormatter: GfxFormatter):
+		assert len(self.celTriLists) == 0
```

Comment:
is there any reason why vertex UV scrolling wouldn't work?

Reply:
To be honest I don't understand what this is, so I didn't try to implement compatibility with it. The way scrolling textures is done in OoT involves updating the tile definition every frame; any UV-based scheme would not be able to independently scroll both textures in a multitexture setup. Is this UV scrolling an SM64 thing or is it something made up by the community?